common/predictive-text/virtualizedWorker.ts

```diff
@@ -0,0 +1,86 @@
+/// <reference path="embedded_worker.d.ts" />
+/// <reference path="worker-interface.d.ts" />
+
+namespace com.keyman.text.prediction {
+  // Unfortunately, importing like this loses all type information for the modules.
+  // It's that, or massively overhaul lm-layer to be module-based instead of
+  // namespace-based.
+  var fs = require("fs");
+  var vm = require("vm");
+
+  class VirtualizedWorkerContext {
+    // The LMLayerWorker installs itself to 'self', the expected Worker global, so we provide an alias.
+    self: VirtualizedWorkerContext;
+
+    constructor() {
+      this.self = this;
+    }
+
+    postMessage: (message: any, options: PostMessageOptions) => void;
```

Comment:
Is `options` ever used? If not, I suggest removing it from the type signature for now:

```suggestion
    postMessage: (message: any) => void;
```